### PR TITLE
statistics: avoid allocating the memory when to auto analyze with pseudo table

### LINKED ISSUE #50100: avoid allocating the memory when to auto analyze with pseudo table
## Enhancement

7lrCfcUNzi

when we meet the many pseudo tables to check whether to do auto-analyze. we will create many pseudo statistics.  it will lead to frequent gc.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #50100

Problem Summary:

### What changed and how does it work?

7lrCfcUNzi

when we meet the many pseudo tables to check whether to do auto-analyze. we will create many pseudo statistics.  it will lead to frequent gc.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/autoanalyze/autoanalyze.go [hunk 1]
```diff
@@ -494,7 +494,7 @@ func tryAutoAnalyzeTable(
 	// 1. If the stats are not loaded, we don't need to analyze it.
 	// 2. If the table is too small, we don't want to waste time to analyze it.
 	//    Leave the opportunity to other bigger tables.
-	if statsTbl.Pseudo || statsTbl.RealtimeCount < AutoAnalyzeMinCnt {
+	if statsTbl == nil || statsTbl.RealtimeCount < AutoAnalyzeMinCnt || statsTbl.Pseudo {
```
Comment (hi-rustin) [inline pkg/statistics/handle/autoanalyze/autoanalyze.go:497]: Please update the comments to explain when the statsTbl is not nil but it is pseudo.

Comment (hawkingrei) [reply pkg/statistics/handle/autoanalyze/autoanalyze.go:497]: I think it is forgotten to delete.

Comment (hawkingrei) [reply pkg/statistics/handle/autoanalyze/autoanalyze.go:497]: It should just change the comment.

**Review by hi-rustin — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**